src/core/lv_obj_pos.c

```diff
@@ -783,15 +788,12 @@ void lv_obj_get_transformed_area(const lv_obj_t * obj, lv_area_t * area, bool re
 {
     lv_point_t p[4] = {
         {area->x1, area->y1},
-        {area->x1, area->y2},
-        {area->x2, area->y1},
-        {area->x2, area->y2},
+        {area->x1, area->y2 + 1},
```

Comment:
Honestly, it's really a mess. E.g. width == 100 means, x1 = 0; x2 = 99. If zoom to 1000% it should be x1 = 0; x2=999.9 (~1000). So as the last pixel is zoomed we need assume that it's there. Combining this with rounding issues due to integer math, it's at least not intuitive. 

It would require a new iteration on the whole transformation logic to make it better somehow.